src/lib_json/json_reader.cpp

```diff
@@ -18,6 +18,8 @@
 #include <memory>
 #include <set>
 #include <limits>
+#include <stdio.h>
```

Comment:
> your great modern compilers which support them without the std:: qualifier are leading you astray

Some older compilers have problems with `std::` sometimes, and `sscanf` in particular can be available in weird ways. We'll be standards compliant in `2.0`, but for now it's helpful to some users if we avoid being too strict.

> I'll let you think on this a while before I do any reverting.

Don't revert. But can you wrap the new include lines in and `ifdef` for your compiler? IIRC, some modern compilers actually do not include the old headers, or consider them a compiler warning (which becomes an error with `-Werror`).

> For instance, I fixed json_writer.cpp in a different way by adding six std::

Could you show us where? We can't add that on `0.y.z`, but on `master` I think that would be fine.